pkg/prometheus/resource_selector.go

```diff
@@ -900,7 +900,35 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 }
 
 func (rs *ResourceSelector) validateKubernetesSDConfigs(ctx context.Context, sc *monitoringv1alpha1.ScrapeConfig) error {
+	promVersion := operator.StringValOrDefault(rs.p.GetCommonPrometheusFields().Version, operator.DefaultPrometheusVersion)
+	version, err := semver.ParseTolerant(promVersion)
+	if err != nil {
+		return fmt.Errorf("failed to parse Prometheus version: %w", err)
+	}
+
 	for i, config := range sc.Spec.KubernetesSDConfigs {
+		if len(config.Selectors) > 0 && !version.GTE(semver.MustParse("2.17.0")) {
+			return fmt.Errorf("selectors are only supported for Prometheus version >= 2.17.0")
+		}
+
+		if config.Role == monitoringv1.RoleEndpointSlice && !version.GTE(semver.MustParse("2.21.0")) {
+			return fmt.Errorf("role endpointslice is only supported for Prometheus version >= 2.21.0")
+		}
+
+		if config.AttachMetadata != nil {
+			if config.Role == monitoringv1.RoleNode || config.Role == monitoringv1.RoleIngress || config.Role == monitoringv1.RoleService {
+				return fmt.Errorf("attachMetadata is only configurable for role Endpoint, EndpointSlice or Pod")
+			}
+
+			if config.Role == monitoringv1.RoleEndpointSlice || config.Role == monitoringv1.RoleEndpoint && !version.GTE(semver.MustParse("2.37.0")) {
+				return fmt.Errorf("attachMetadata for the role EndpointSlice and role Endpoint is only supported for Prometheus version >= 2.37.0")
+			}
+
+			if config.Role == monitoringv1.RolePod && !version.GTE(semver.MustParse("2.35.0")) {
```

Comment:
```suggestion
			if config.Role == monitoringv1.RolePod && version.LT(semver.MustParse("2.35.0")) {
```

Reply:
Do you think it would it be better to keep it `GTE`, because we do that in other areas of the code? or it won't make a big difference?